quark/radiocontrast.py

```diff
@@ -0,0 +1,138 @@
+# This file is part of Quark-Engine - https://github.com/quark-engine/quark-engine
+# See the file 'LICENSE' for copying permission.
+
+import os
+import json
+
+from tqdm import tqdm
+from quark.core.quark import Quark
+from quark.core.struct.ruleobject import RuleObject
+
+
+class RadioContrast:
+    """
+    This module is for generating rules with the APIs in a specific method.
+    """
+
+    def __init__(self, apk_path, target_method, output_dir):
+        self.quark = Quark(apk_path)
+        self.apkinfo = self.quark.apkinfo
+
+        # Parse smali code into classname methodname and descriptor.
+        classname = target_method.split("->")[0]
+        methodname = target_method.split("->")[1].split("(")[0]
+        descriptor = "(" + target_method.split("->")[1].split("(")[1]
+
+        self.method = self.apkinfo.find_method(
+            class_name=classname, method_name=methodname, descriptor=descriptor,
+        )
+
+        if self.method is None:
+            print("ERROR: start method or end method not found")
+            return
+
+        self.output_dir = output_dir
+        self.api_set = set()
+        self.MAX_SEARCH_LAYER = 3
```

Comment:
I suggest adding a default argument to the method `__init__`. In this way, users can change the value of the variable MAX_SEARCH_LAYER if they want.